openmeter/ent/schema/productcatalog.go

```diff
@@ -0,0 +1,251 @@
+package schema
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+
+	"entgo.io/ent"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/dialect/sql"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/openmeter/productcatalog/plan"
+	"github.com/openmeterio/openmeter/pkg/datex"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type Plan struct {
+	ent.Schema
+}
+
+func (Plan) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.KeyMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+		entutils.MetadataAnnotationsMixin{},
+	}
+}
+
+func (Plan) Fields() []ent.Field {
+	return []ent.Field{
+		field.Int("version").
+			Min(1),
+		field.String("name").
+			NotEmpty(),
+		field.String("description").
+			Optional().
+			Nillable(),
+		field.String("currency").
+			Default("USD").
+			NotEmpty().
+			Immutable(),
+		field.Time("effectiveFrom").
```

Comment:
we usually use underscore to define ent schema

Reply:
Agreed. Copy/paste error. Will fix it now.